File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
const maxProfit = (prices) => {
    let buy = prices[0];
    let profit = 0;
    
    for(let i= 0 ; i < prices.length; i++){
        if(buy >  prices[i]){
            buy = prices[i]
        }else{
            let prof = prices[i] - buy
            console.log(buy)
            profit = Math.max(prices[i] - buy, profit )
        }
    }
    return profit
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        output = 0
        l = 0
        
        for r in range(1, len(prices)):
            if prices[r] < prices[l]:
                l = r
            output = max(output, prices[r] - prices[l])

            
        return output